### PR TITLE
LOOP-1717: Swap new SettingsView for SettingsTableViewController (part 1 of N)

### PR DESCRIPTION
**Pull request by rickpasetto**
This is the next step in swapping out the old settings UI for the new one.  Now, tapping the gear icon will launch the new `SettingsView` instead of `SettingsTableViewController`.  Navigation is still a bit strange; some screens that `SettingsView` navigates to still need it to "dismiss" before opening.  That will be addressed in the coming steps.

### CODE REVIEW

**Review by novalegra — COMMENTED** [review]
Looks good! 

One q - I'm assuming it's intentional that there are no services in the settings screen at the moment? Or that I'm just not clever enough to find it? If there's no way to access the Tidepool service in there and this is merged to dev, then QA won't be able to test acceptance-flow-related work

**Comment by rickpasetto** [comment]
@novalegra That will be "turned back on" by a FeatureFlag (see https://github.com/tidepool-org/LoopWorkspace/pull/222)

File: Loop/View Controllers/StatusTableViewController.swift [hunk 1]
```diff
@@ -1652,3 +1738,96 @@ fileprivate extension UIViewController {
...
+        case .carbRatio:
+            deviceManager?.loopManager.carbRatioSchedule = therapySettings.carbRatioSchedule
+            deviceManager?.analyticsServicesManager.didChangeCarbRatioSchedule()
+        case .insulinSensitivity:
+            deviceManager?.loopManager.insulinSensitivitySchedule = therapySettings.insulinSensitivitySchedule
+            deviceManager?.analyticsServicesManager.didChangeInsulinSensitivitySchedule()
+        case .none:
+            break // NO-OP
+        }
+    }
```
Comment (nhamming) [inline Loop/View Controllers/StatusTableViewController.swift:1778]: [nit] I wonder why the `StatusTableViewController` needs to handle this logic. Curious if providing the `deviceManager` to the `SettingsViewModel` and letting the view model handle this logic might be a more appropriate approach?

**Review by nhamming — APPROVED** [review]
Just a question about where settings logic should reside. Otherwise LGTM.

**Review by novalegra — APPROVED** [review]

File: Loop/View Controllers/StatusTableViewController.swift (hunk 1 above)
Comment (rickpasetto) [reply Loop/View Controllers/StatusTableViewController.swift:1778]: @nhamming I hear you... but I'm not a fan of passing around big classes like `DeviceDataManager` everywhere.  That makes it very hard to test.  I may create a protocol that contains `didSave()` and give that to `SettingsViewModel`, and implement this logic in `DeviceDataManager`.  But I consider that a later refactor.

**Pull request merged**